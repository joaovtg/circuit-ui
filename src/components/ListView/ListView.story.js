import React, { Component } from 'react';
import styled from 'react-emotion';
import { range } from 'lodash/fp';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
// import { action } from '@storybook/addon-actions';
import * as knobs from '@storybook/addon-knobs/react';

import withTests from '../../util/withTests';
import List from '.';

const Item = styled(List.Item)`
  min-width: 320px;
`;

class ListContainer extends Component {
  state = { selected: 0 };

  handleClick = selected => () => this.setState({ selected });

  render() {
    const { selected } = this.state;
    const padding = knobs.select(
      'padding',
      [List.Item.KILO, List.Item.MEGA, List.Item.GIGA],
      List.Item.GIGA
    );

    return (
      <List>
        {range(1, 6).map(i => (
          <Item
            key={i}
            selected={selected === i}
            onClick={this.handleClick(i)}
            padding={padding}
          >
            Item #{i}
          </Item>
        ))}
      </List>
    );
  }
}

storiesOf('ListView', module)
  .addDecorator(withTests('ListView'))
  .add('ListView', withInfo()(() => <ListContainer />));
