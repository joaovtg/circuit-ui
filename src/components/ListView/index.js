import Wrapper from './components/Wrapper';
import Item from './components/Item';

Wrapper.Item = Item;

export default Wrapper;
