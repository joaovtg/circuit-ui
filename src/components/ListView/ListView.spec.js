import React from 'react';

import ListView from '.';

const BaseList = () => (
  <ListView>
    <ListView.Item>List item 1</ListView.Item>
    <ListView.Item selected>List item 2</ListView.Item>
    <ListView.Item>List item 3</ListView.Item>
    <ListView.Item>List item 4</ListView.Item>
  </ListView>
);

describe('ListView', () => {
  /**
   * Style tests.
   */
  it('should render with default styles', () => {
    const actual = create(<BaseList />);
    expect(actual).toMatchSnapshot();
  });

  /**
   * Accessibility tests.
   */
  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<BaseList />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
