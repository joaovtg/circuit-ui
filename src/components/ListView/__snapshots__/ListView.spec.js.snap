// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView should render with default styles 1`] = `
.circuit-16 {
  background-color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  box-shadow: 0 0 0 1px rgba(12,15,20,0.02), 0 0 1px 0 rgba(12,15,20,0.06), 0 2px 2px 0 rgba(12,15,20,0.06);
  padding: 16px 24px;
  margin: 0 0 12px 0;
  padding: 0;
}

.circuit-16 > .circuit-3 {
  border-bottom: 1px solid #D8DDE1;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.circuit-16 > .circuit-3:last-child {
  border-bottom: none;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.circuit-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  position: relative;
  cursor: pointer;
  padding: 24px;
}

.circuit-2:hover .circuit-1 {
  display: block;
}

.circuit-0 {
  display: none;
  box-shadow: 0px 0px 0px 2px #3388FF;
  border-radius: 4px;
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  z-index: 1;
}

.circuit-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  position: relative;
  cursor: pointer;
  padding: 24px;
  background: #DAEAFF;
}

.circuit-6:hover .circuit-1 {
  display: block;
}

<div
  className="circuit-16 circuit-17"
  spacing="giga"
>
  <div
    className="circuit-2 circuit-3"
    selected={false}
  >
    List item 1
    <div
      className="circuit-0 circuit-1"
    />
  </div>
  <div
    className="circuit-6 circuit-3"
    selected={true}
  >
    List item 2
    <div
      className="circuit-0 circuit-1"
    />
  </div>
  <div
    className="circuit-2 circuit-3"
    selected={false}
  >
    List item 3
    <div
      className="circuit-0 circuit-1"
    />
  </div>
  <div
    className="circuit-2 circuit-3"
    selected={false}
  >
    List item 4
    <div
      className="circuit-0 circuit-1"
    />
  </div>
</div>
`;
